Update docs to remove auto-tables (#6874)
cakephp/docs: en/orm.rst

Quick Example
=============

To get started you don't have to write any code. If you've followed the
:ref:`CakePHP conventions for your database tables
<model-and-database-conventions>` you can just start using the ORM. For example
if we wanted to load some data from our ``articles`` table we would start off

> creating our ``Articles`` table class. Add the following to
> **src/Model/Table/ArticlesTable.php** after the ``<?php`` opening tag::

Reviewer: ```suggestion
creating our ``Articles`` table class. Create
**src/Model/Table/ArticlesTable.php** with the following code::
```

    <?php
    namespace App\Model\Table;

    use Cake\ORM\Table;

    class ArticlesTable extends Table
    {
    }

> Then in a controller, or command we can have CakePHP create an instnace for us::

Reviewer 2: ```suggestion
Then in a controller, or command we can have CakePHP create an instance for us::
```

> Then in a controller, or command we can have CakePHP create an instance for us::

Reviewer: ```suggestion
Then in a controller or command we can have CakePHP create an instance for us::
```

    public function someMethod()
    {
        $this->loadModel('Articles');
        $resultset = $this->Articles->find()->all();

        foreach ($resultset as $row) {
            echo $row->title;
        }
    }

> In other contexts, you can use the ``LocatorAwareTrait`` to add methods that
> enable you to get ORM tables::

Reviewer: ```suggestion
In other contexts, you can use the ``LocatorAwareTrait`` which add accessor methods for ORM tables::
```

    use Cake\ORM\Locator\LocatorAwareTrait;

    public function someMethod()
    {
        $articles = $this->getTableLocator()->get('Articles');
        // more code.
    }

Within a static method you can use the :php:class:`~Cake\\Datasource\\FactoryLocator` 
to get the table locator::

>     $articles = FactoryLocator::get('Table')->get('Articles');

Reviewer 2: This should be using TableRegistry::getTableLocator()-> instead IMO.
The magic string one is a bit weird for app level usage.

Author: This example has been in the docs for a while.

Reviewer 2: Yeah, i still think its wrong to leak that internal detail.
